tests/unit/test_powerwall.py

```diff
@@ -39,52 +41,51 @@
 )
 
 
-class TestPowerWall(unittest.TestCase):
-    def setUp(self):
+class TestPowerWall(unittest.IsolatedAsyncioTestCase):
+    async def asyncSetUp(self):
+        self.aresponses = aresponses.ResponsesMockServer()
+        await self.aresponses.__aenter__()
+
         self.powerwall = Powerwall(ENDPOINT)
 
+    async def asyncTearDown(self):
+        await self.powerwall.close()
+        await self.aresponses.__aexit__(None, None, None)
+
     def test_get_api(self):
         self.assertIsInstance(self.powerwall.get_api(), API)
 
-    @responses.activate
-    def test_get_charge(self):
-        add(
-            Response(
-                GET,
-                url=f"{ENDPOINT}system_status/soe",
-                json={"percentage": 53.123423},
-            )
-        )
-        self.assertEqual(self.powerwall.get_charge(), 53.123423)
-
-    @responses.activate
-    def test_get_sitemaster(self):
-        add(
-            Response(
-                responses.GET,
-                url=f"{ENDPOINT}sitemaster",
-                json=SITEMASTER_RESPONSE,
-            )
+    def add_response(self, path: str, method: str = "GET", json: object = None):
```

Comment:
```suggestion
    def add_response(self, path: str, method: str = "GET", body: Optional[dict] = None):
```
I would prefer to not use the name of the `json`, and instead use something else